text/XXXX-externalize_private_keys.md

```diff
@@ -0,0 +1,111 @@
+- Feature Name: Externalize Private Keys
+- Start Date: <!-- (fill me in with today's date, YYYY-MM-DD) -->
+- RFC PR: <!-- (leave this empty) -->
+- Issue: <!-- (leave this empty) -->
+
+# Summary
+[summary]: #summary
+
+Separate Qri user private key from configuration, storing in the operating system's keyring, or superceeded by an envrionrment variable `QRI_PRIVATE_KEY`.
+
+# Motivation
+[motivation]: #motivation
+
+Qri users are getting themselves into un-recoverable states by manually deleting their `.qri` directory, which takes their private key with it, creating major problems when they try to re-add & change datasets. Because Qri is built on public-key infrastructure (PKI), deleting the private key has the effect of locking a user out of their own identity.
+
+More importantly, storing private keys in configuration is a security vulnerability. Coupling sensitive information with configuration renders the entire configuration sensitive.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+We inherited this bad habit of storing private keys in configuration as a "deal with it later" approach taken from IPFS, unlike Qri, IPFS identities are not "significant" in the sense that it's not tied to an identity intended for humans. Tying an identity to a private key means we need to deal with this problem much earlier.
+
+This RFC deprecates `profile.privKey` config value, moving the the private key to the user's operating system keyring. When no keyring is available, store the keyring in a seperate file called `QRI_PRIVATE_KEY` within the `.qri` directory, and present the user with a warning that they should copy this file to a safe location.
+
+In all cases the active private key can be overridden with an environment variable: `QRI_PRIVATE_KEY`. In all cases the key is expected to be a base64-encoded string.
```

Comment:
It would be nice to be able to support `qriswitch` still, to support multiple profiles that the user can switch between. The environment variable helps, but we should also store the users keys at multiple locations then, which is why I recommend having a subdirectory named after the `profileID` above. Using only a single file makes that much harder to do.

Also, using only a single file makes it easier to lose data if a user deletes their identity and makes a new one multiple times in a row. Keeping everything indexed by profileID avoids this problem.

Reply:
+1 - also think the `.ssh` thing complements this further

Reply:
Yeah I think we'll always need to support both. Env var for the "master switch". 

Having directories for profiles is a very useful arrangement. I think we should consider that setup for the `.qri` directory itself.